### ISSUE TITLE
Zoomable light table on max zoom 1 is very slow or doesn't show them at all

### ISSUE THREAD

**Issue by maboleth** [darktable-org/darktable#5924]
Zooming the image in zoomable light table mode not only never centers the image on the screen (like on 3.0.2) but sometimes never displays it. The workaround is to 'tilt' the scrolling or move the mouse.

Please check the screencast video:
[dt.zip](https://github.com/darktable-org/darktable/files/5058774/dt.zip)

**Comment by maboleth** [comment]
Zooming in fast results in many images not shown on max screen zoom. As seen in those screenshots.

I have also noticed that in order for images to reappear on their placements, you need to be close to the upper left angle. Because apparently in zoomable mode, images start as small thumbnails that expand. 

And the trigger to expand them on the screen is upper left corner. 

![Untitled2](https://user-images.githubusercontent.com/2467520/90128301-f8469780-dd66-11ea-94c3-71a81dc6dc94.jpg)
![Untitled](https://user-images.githubusercontent.com/2467520/90128304-f8df2e00-dd66-11ea-9be2-35e98c673fb7.jpg)

If you scroll "under" those thumbnails (under upper left corner), images won't be shown. As seen on this screen cast.

[scast.zip](https://github.com/darktable-org/darktable/files/5068712/scast.zip)

**Comment by maboleth** [comment]
I have at least got a confirmation from one user that this works at his end without problems.

Here's another video that really shows how this issue behaves. Please check it out:
[scast-bug.zip](https://github.com/darktable-org/darktable/files/5073938/scast-bug.zip)

So, apparently, something is causing this delay on my machine.

Hardware:
AMD Ryzen 3700x, 32gb ram, nvme SSD, Nvidia 1060 6GB DDR5 GPU. All drivers are latest, Arch Linux with Gnome (all latest, as it always is with Arch).

Disabling OpenGL does not improve this (quite the contrary, but at least I ruled that out). I have enabled thumbnail & full preview caches.

If any of the devs could help me troubleshoot this, I'd appreciate it a lot.

**Comment by maboleth** [comment]
The guy tested it and this is his findings. 

_dt 3.3.0+88~g79ac4657c, Ubuntu 18.04
I just tried again on an other (less powerful) machine.

    Here I can reproduce the effect of getting an “empty” image area when in “zoomable lighttable” at size 1. The image area is not really empty, it is filled with the rectangles for the thumbnails but the images inside are smaller than expected, positioned at the upper left corners of the rectangles. The thumbnails with the correct size are filled in, if one moves around.
    After switching from “zoomable lighttable” to “file manger” and back again the effect can not be reproduced again. So caching seems to have a markedly influence.
    I also could reproduce the effect to get an image area which is completely empty. This happens in “zoomable lighttable” if the center of the image area is empty (all thumbnails significant above the center), and one then is zooming in, until all thumbnails are outside (above) the image area. If you then zoom back, you don’t get back the thumbnails. One could call this a bug. But there are two very simple workarounds :
    group your thumbnails around the center of the image area when zooming in or
    switch to “file manager” and back to “zoomable lighttable” and the thumbnails are back.

In conclusion : The effect obviously depends on the hardware used and caching of the thumbnails. With a powerful hardware and thumbnails precalculated it is possible to get a very smooth workflow. There are some smaller issues (described above) which can be circumvented easily._

**Comment by esm7** [comment]
Suffering from this too. Dell XPS 15 7590 running Manjaro Linux and i3wm.

**Comment by maboleth** [comment]
Hope we will get some support from the devs, at least to troubleshoot some more.

My machine is anything but slow - ryzen 3700x, 32gb ram, nvme ssd, nvidia 1060 gpu with 6gb ddr5.

**Comment by helloworld1** [comment]
That exactly describes the problem. Mine is Arch Linux + Wayland + Fast AMD CPU / GPU. DT 3.0.1 is blazingly fast in zoomable light table. But DT 3.2.1, the performance is slow, and zoom to 1x does not actually full screen the image correctly.

**Comment by matt-maguire** [comment]
Wayland doesn’t support colour management. What happens if you switch to Xorg/X11?

**Comment by maboleth** [comment]
I'm using X11 already, no Wayland. But some or most of these issues are fixed in master and others didn't provide any more useful info, so I will close this report. Thanks.
